mlrun/__main__.py

```diff
@@ -372,6 +378,9 @@ def run(
     set_item(runobj.spec.hyper_param_options, selector, "selector")
 
     set_item(runobj.spec, inputs, run_keys.inputs, list2dict(inputs))
+    set_item(
+        runobj.spec, returns, run_keys.returns, [py_eval(value) for value in returns]
```

Comment:
I thought it is being passed as strings, so `py_eval` would return me a dictionary from the string in case the user passed a dictionary. In addition, I saw it is being used to parse each parameter (`-p x`) to construct the `params` dictionary so I used it as well. The log hints can be a string with white spaces or a dictionary, the dictionary if I understand the transformation is being parsed to string and then concatenated to the mlrun command, so this is why I used `py_eval`.

Reply:
`py_eval` uses `ast.literal_eval` which is a general utility to translate strings into py objects. It will work here, but it's an overkill for your purposes - you could just `json.loads()` which will be enough for your purposes.